### DISCUSSION TITLE
Problem with Calendar component in Web component

### DISCUSSION THREAD

**Discussion by pearl1488** [primefaces/community#1464] [PrimeNG Templates]
I've encountered an issue with an Angular component, specifically the p-calendar, which I've converted into a web component using Angular elements. The problem arises when selecting a date from the picker or navigating to the next month, as the popup closes unexpectedly without properly binding the selected data. This issue is isolated to the Angular web component.

Could you help me to resolve the issue?

![image](https://github.com/primefaces/community/assets/37859116/708ea7d9-a620-47a8-a7de-79ea24e09dc8)
![image](https://github.com/primefaces/community/assets/37859116/50bb96b7-6aa5-4f85-8315-df05cfea6fc8)
![image](https://github.com/primefaces/community/assets/37859116/87d544c9-b96a-4547-a007-3b35f30a8d9c)

**Comment by cetincakiroglu** [comment]
Hi,

Could you please create a stackblitz example so we can investigate and identify the issue?

**Reply by pearl1488** [reply]
Thanks for your response. Please take a look at the stack blitz below. In my case, I should use shadow dom.
https://stackblitz.com/edit/stackblitz-starters-4ocgzn?file=src%2Fapp%2Fdatapicker%2Fdatapicker.component.ts
